### PR TITLE
add switch component

### PR DESCRIPTION
**Pull request by hamilton**
This component is mean to be used in a few places:
- @djbarnwal we'd like to replace the include / exclude toggle in the filter pill menu with this + the word "Exclude". Off means include, on means exclude
- this component will be useful in many other places within the UI, and specifically for the upcoming period of period comparison functionality.

Kapture 2022-10-22 at 07 29 05

### CODE REVIEW

**Comment by bcolloran** [comment]
looks good to me, the only thing I'll mention is that, I would recommend having the active state Blue 400, as depicted here https://www.figma.com/file/wr73rSbI70HNrGNKocoBHy/Rill-Application-Design?node-id=3677%3A38874 -- I think it's consistent with the other uses of blue and adds just a little more life and interest to the app, as well as adding a bit of additional contrast between the states. I believe that using color for the active state is fairly conventional, even for apps that make fairly restrained use of color like we do what -- for example here is what Figma does

image
image




But this is a small point. If you want to go with the grayscale version I think that's fine

**Review by bcolloran — APPROVED** [review]

**Comment by hamilton** [comment]
Yep, seems sensible. Best to utilize `accent-color` here I think (and also best to set `accent-color`)

File: web-local/src/lib/components/button/Switch.svelte [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  type="checkbox"
+  {id}
+/>
+
+<style lang="postcss">
+  input {
+    @apply bg-gray-400;
+    --width: 22px;
+    --height: 12px;
+    --margin: 3px;
```
Comment (djbarnwal) [inline web-local/src/lib/components/button/Switch.svelte:23]: nit: The placement of the circle and animation looks a bit weird for me in Firefox (105).
Kapture 2022-10-23 at 12 22 11

Playing around with `--margin: 2px` it looks better. Although `3px` seems to look good in Chrome.

**Comment by hamilton** [comment]
Marissa took a look and we struggled to find any visual difference b/t FF, Chrome, and Safari. If we can reproduce we will mak ea fix, but I think we should merge

**Pull request merged**